src/main/java/com/google/cloud/tools/maven/AppEngineFactory.java

```diff
@@ -27,17 +27,38 @@
  */
 public interface AppEngineFactory {
 
+  /**
+   * Supported dev app server versions.
+   */
+  public enum SupportedVersion {
```

Comment:
since this isn't scoped to the server explicitly?, should it SupporterServerVersion?